File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03944/p03944_num2_parsed.cpp
#include <iostream>

using namespace std;

int main() {
    int W, H, N;
    cin >> W >> H >> N;

    // Initialize the white rectangle area
    int whiteArea = W * H;

    // Store the coordinates and actions
    int x[N], y[N], a[N];
    for (int i = 0; i < N; i++) {
        cin >> x[i] >> y[i] >> a[i];
    }

    // Iterate through each point
    for (int i = 0; i < N; i++) {
        // Adjust the white area based on the action
        switch (a[i]) {
            case 1:
                whiteArea -= x[i] * H;
                break;
            case 2:
                whiteArea -= (W - x[i]) * H;
                break;
            case 3:
                whiteArea -= W * y[i];
                break;
            case 4:
                whiteArea -= W * (H - y[i]);
                break;
        }
    }

    cout << whiteArea << endl;

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03944/p03944_num2_parsed.py
W, H, N = map(int, input().split())

# Initialize the white region area
area = W * H

# Iterate through each point
for _ in range(N):
    x, y, a = map(int, input().split())
    
    # If a_i = 1, reduce the white region area in the left of x
    if a == 1:
        area -= x * H
    
    # If a_i = 2, reduce the white region area in the right of x
    elif a == 2:
        area -= (W - x) * H
    
    # If a_i = 3, reduce the white region area below y
    elif a == 3:
        area -= y * W
    
    # If a_i = 4, reduce the white region area above y
    elif a == 4:
        area -= (H - y) * W

# Print the remaining white region area
print(area)